aten/src/ATen/native/SoftMax.cpp

```diff
@@ -418,6 +418,11 @@ Tensor softmax(const Tensor& input_, const int64_t dim_, c10::optional<ScalarTyp
   return result;
 }
 
+Tensor& softmax_out(const Tensor& input_, const int64_t dim_, c10::optional<ScalarType> dtype, Tensor& output_) {
+  Tensor converted = dtype.has_value() ? input_.toType(dtype.value()) : input_;
```

Comment:
should you have the same logic to handle fp16->fp32 conversion on cuda that regular softmax/logsoftmax has?

Reply:
I am not familiar with CUDA. Do you know anyone who I can talk to?

Reply:
Ahh okay i think i know what you mean